source/core/lethe_grid_tools.cc

```diff
@@ -1785,3 +1786,34 @@ LetheGridTools::find_line_sphere_intersection(
   const Tensor<1, 3> &line_direction,
   const Point<3>     &sphere_center,
   const double       &sphere_radius);
+
+template <int dim>
+void
+LetheGridTools::rotate_mapping(const DoFHandler<dim> &dof_handler,
+                               const MappingQ<dim>   &mapping,
+                               const double           radius,
+                               const double           rotation_angle)
+{
+  // mapping cache object
+  MappingQCache<dim> mapping_cache(mapping.get_degree());
+
+  mapping_cache.initialize(
+    mapping,
+    dof_handler.get_triangulation(),
+    [&](const auto &cell, const auto &point) {
+      if (cell->center().norm() > radius)
+        return point;
+
+      return static_cast<Point<dim>>(
+        Physics::Transformations::Rotations::rotation_matrix_2d(
+          rotation_angle) *
+        point);
+    },
+    false);
```

Comment:
Are you sure this work? mappingQCache is declared locally, won't it go out of scope and be de-allocated after the function?.
Have you tested that function in anyway?